esp_wrover_kit: hello_world build failure

**Describe the bug**
`samples/hello_world` (and all builds I tried) fail for the new `esp_wrover_kit` board with a CMake warning. I'm hoping this isn't some kind of crazy weird issue with my environment since I'm able to build for the `esp32` just fine.

CMake warning:
```
CMake Error at ../../cmake/extensions.cmake:2896 (message):
  dt_reg_addr(img_0_off ...) missing required argument: PATH
Call Stack (most recent call first):
  ../../soc/xtensa/esp32/CMakeLists.txt:13 (dt_reg_addr)
```
`../../soc/xtensa/esp32/CMakeLists.txt:12-13`:
```
dt_nodelabel(dts_partition_path NODELABEL "slot0_partition")
dt_reg_addr(img_0_off PATH ${dts_partition_path})
```

Poking around in my build folder, I noticed that `zephyr.dts` made no mention of `boot_partition`, `slot0_partition`, etc. listed in `soc/xtensa/esp32/CMakeLists.txt`, only the `storage_partition` already defined in `boards/xtensa/esp_wrover_kit/esp_wrover_kit.dts`. These are, of course, defined in `boards/xtensa/esp32/esp32.dts`. I'm not sure why they're not coming in.

**To Reproduce**

`west build -p -b esp_wrover_kit samples/hello_world`

**Expected behavior**
Build success.

**Impact**
Showstopper

**Logs and console output**
```
 $ west build -p -b esp_wrover_kit samples/hello_world/
-- west build: making build dir /home/herbert/zephyr-external/zephyr/build pristine
-- west build: generating a build system
Including boilerplate (Zephyr base): /home/herbert/zephyr-external/zephyr/cmake/app/boilerplate.cmake
-- Application: /home/herbert/zephyr-external/zephyr/samples/hello_world
-- Zephyr version: 2.7.99 (/home/herbert/zephyr-external/zephyr), build: v2.7.99-3416-g7dac931e3662
-- Found Python3: /usr/bin/python3.7 (found suitable exact version "3.7.5") found components: Interpreter 
-- Found west (found suitable version "0.12.0", minimum required is "0.7.1")
-- Board: esp_wrover_kit
-- Cache files will be written to: /home/herbert/.cache/zephyr
-- Found host-tools: zephyr 0.13.1 (/home/herbert/zephyr-sdk-0.13.1)
-- Found dtc: /home/herbert/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6")
-- Found BOARD.dts: /home/herbert/zephyr-external/zephyr/boards/xtensa/esp_wrover_kit/esp_wrover_kit.dts
-- Generated zephyr.dts: /home/herbert/zephyr-external/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /home/herbert/zephyr-external/zephyr/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /home/herbert/zephyr-external/zephyr/build/zephyr/include/generated/device_extern.h
-- Including generated dts.cmake file: /home/herbert/zephyr-external/zephyr/build/zephyr/dts.cmake
Parsing /home/herbert/zephyr-external/zephyr/Kconfig
Loaded configuration '/home/herbert/zephyr-external/zephyr/boards/xtensa/esp_wrover_kit/esp_wrover_kit_defconfig'
Merged configuration '/home/herbert/zephyr-external/zephyr/samples/hello_world/prj.conf'
Configuration saved to '/home/herbert/zephyr-external/zephyr/build/zephyr/.config'
Kconfig header saved to '/home/herbert/zephyr-external/zephyr/build/zephyr/include/generated/autoconf.h'
-- Found toolchain: espressif (/home/herbert/.espressif/tools/zephyr)
-- The C compiler identification is GNU 8.4.0
-- The CXX compiler identification is GNU 8.4.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/herbert/.espressif/tools/zephyr/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc
CMake Error at ../../cmake/extensions.cmake:2896 (message):
  dt_reg_addr(img_0_off ...) missing required argument: PATH
Call Stack (most recent call first):
  ../../soc/xtensa/esp32/CMakeLists.txt:13 (dt_reg_addr)


-- Configuring incomplete, errors occurred!
See also "/home/herbert/zephyr-external/zephyr/build/CMakeFiles/CMakeOutput.log".
FATAL ERROR: command exited with status 1: /home/herbert/.local/bin/cmake -DWEST_PYTHON=/usr/bin/python3.7 -B/home/herbert/zephyr-external/zephyr/build -S/home/herbert/zephyr-external/zephyr/samples/hello_world -GNinja -DBOARD=esp_wrover_kit -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
```

**Environment**
 - OS: Ubuntu 18.04.6
 - Toolchain: Espressif
 - Commit: 7dac931 (latest)
 - SDK: 0.13.2
 - CMake: 3.20.5

[build.zip](https://github.com/zephyrproject-rtos/zephyr/files/7893727/build.zip)

--- a/boards/xtensa/esp_wrover_kit/esp_wrover_kit.dts
+++ b/boards/xtensa/esp_wrover_kit/esp_wrover_kit.dts
@@ -137,9 +137,34 @@
 		#address-cells = <1>;
 		#size-cells = <1>;
 
-		storage_partition: partition@9000 {
+		/* Reserve 60kB for the bootloader */
+		boot_partition: partition@1000 {
+			label = "mcuboot";
+			reg = <0x00001000 0x0000F000>;
+			read-only;
+		};
+
+		/* Reserve 1024kB for the application in slot 0 */
+		slot0_partition: partition@10000 {
+			label = "image-0";
+			reg = <0x00010000 0x00100000>;
+		};
+
+		/* Reserve 1024kB for the application in slot 1 */
+		slot1_partition: partition@110000 {
+			label = "image-1";
+			reg = <0x00110000 0x00100000>;
+		};
+
+		/* Reserve 256kB for the scratch partition */
+		scratch_partition: partition@210000 {
+			   label = "image-scratch";
+			   reg = <0x00210000 0x00040000>;
+		};
+
+		storage_partition: partition@250000 {
 			label = "storage";
-			reg = <0x00009000 0x00006000>;
+			reg = <0x00250000 0x00006000>;
 		};
 	};
 };

